RecoEgamma/EgammaElectronProducers/plugins/TrackingRegionsFromSuperClustersProducer.h

```diff
@@ -147,9 +154,16 @@ TrackingRegionsFromSuperClustersProducer(const edm::ParameterSet& cfg,
   deltaPhiRegion_        = regionPSet.getParameter<double>("deltaPhiRegion");
   deltaEtaRegion_        = regionPSet.getParameter<double>("deltaEtaRegion");
   useZInVertex_          = regionPSet.getParameter<bool>("useZInVertex");
+  useZInBeamspot_        = regionPSet.getParameter<bool>("useZInBeamspot");
+  useDefaultZ_           = regionPSet.getParameter<bool>("useDefaultZ");
```

Comment:
useDefaultZ_ is never used as such in the code, and it looks redundant here. And, by the way, it is not even the "default", as it is false by default in the fillDescriptions method.

If you get rid of it, the check implemented now in validateConfigSettings() can simply become

   if (useZInVertex_ && useZInBeamspot_) throw ...

And, to follow the flow that doesn't use neither the vertex or the beamspot one can just (more intuitively) set both those two flags as false.